### PR TITLE
[minipass] Remove types for minipass

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussion: #60874

minipass includes its own types as of version 3.3.1 ([unpkg](https://unpkg.com/browse/minipass@3.3.1/))
Its new types aren't fully compatible with the old ones, but I only had to make a slight modification to the code for one library (minizlib).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by MysteryBlokHed** [comment]
It looks like the types for the latest version of minipass incorrectly use `export default` instead of `export =` syntax. It might be a good idea to keep the DT types up until that's changed

**Comment by MysteryBlokHed** [comment]
Still waiting on isaacs/minipass#41, but I'll update this PR once that one's merged

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## tar ([<kbd>unpkg</kbd>](https://unpkg.com/browse/tar@latest/))
was missing the following properties:
1. Unpack
1. ReadEntry
1. WriteEntry
1. Header
1. Pax

<details>
<summary>as well as these 5 other properties...</summary>
<p>Unpack, ReadEntry, WriteEntry, Header, Pax</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4bc6fb8a47f65d9c0cca4a67da4ade7a92012b3f
</p>

**Comment by peterblazejewicz** [comment]
Minipass types are now source of the truth, let's just wait for update of the merged change.
The version here can stay or can be bumped

**Comment by gabritto** [comment]
@peterblazejewicz @MysteryBlokHed I'm a bit confused, are we still waiting for https://github.com/isaacs/minipass/pull/41 to merge this PR? Because that minipass PR was closed.

**Comment by peterblazejewicz** [comment]
IMO PR was merged, there was no release with a fix

**Comment by gabritto** [comment]
Ok, I see that now. @MysteryBlokHed please let us know here when a minipass release with the export fixes is published, so we can merge this PR. Thanks for your contribution!

**Comment by MysteryBlokHed** [comment]
Weird. minipass release 3.3.5 fixed the issue, but it's not marked as `latest`, so npm shows 3.3.4 on its page instead:

Screenshot of minipass on npm
Screenshot of minipass's active releases

**Comment by MysteryBlokHed** [comment]
@peterblazejewicz @gabritto—The `latest` tag still hasn't been updated, but I don't think it should affect the types, since their dependency is on version `3.3.5`. This should mean that an incompatible version can't be downloaded, so I don't think there'd be a problem with merging this.

File: notNeededPackages.json
Comment (peterblazejewicz) [inline notNeededPackages.json:2910]:
```suggestion
            "asOfVersion": "3.3.5"
```
https://unpkg.com/browse/minipass@3.3.5/index.d.ts

Comment (peterblazejewicz) [reply notNeededPackages.json:2910]: this ships with proper export

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Pull request merged**
